Source/Core/Core/IOS/USB/Emulated/Skylanders/Skylander.cpp

```diff
@@ -1193,7 +1185,7 @@ void SkylanderPortal::QueryBlock(u8 sky_num, u8 block, u8* reply_buf)
   if (skylander.status & 1)
   {
     reply_buf[1] = (0x10 | sky_num);
-    memcpy(reply_buf + 3, skylander.data.data() + (16 * block), 16);
+    skylander.figure->GetBlock(block, reply_buf + 3);
```

Comment:
There's a few places where this is relevant: Is there any guarantee that `figure` is actually set to something? As far as I can tell it's initialized to null and if `LoadSkylander()` has not actually put a figure in that slot then this just dereferences a nullptr.

On a similar note, this isn't technically new, but there also doesn't seem to be any checks for if the slot is still filled after a call to `RemoveSkylander()` would have removed it -- `figure` isn't even reset there, for that matter.

Reply:
The way Skylanders portals, and especially this implementation, work is that when a figure is removed, the portal immediately sends a status update telling the game that the figure has been removed. This means that the game never queries or writes to a Skylander that is not on the portal. I'm going to add some extra checks to Write and Query to handle that case still because a real portal does too

Reply:
There are already status checks where applicable